workspaces/agent-forge/plugins/agent-forge/src/components/ChatAssistantApp.tsx

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import ChatFeedback from './ChatFeedback';
+import ChatHeader from './ChatHeader';
+import ChatInput from './ChatInput';
+import ChatTabs from './ChatTabs';
+import React, { useState, useEffect, useRef, useMemo } from 'react';
+import WebexLogo from '../icons/jarvis.png';
+import useStyles from './useStyles';
+import { ChatSuggestionOptions } from './ChatSuggestionOptions';
+import { Message, Feedback, UserResponse, IQuestionResponse } from '../types';
+import {
+  appThemeApiRef,
+  configApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import {
+  createTimestamp,
+  delay,
+  isAnswer,
+  isSuggestions,
+  makeLinksClickable,
+} from '../utils';
+import { ChatbotApi } from '../apis';
+import { useObservable } from 'react-use';
+import { v4 as uuidv4 } from 'uuid';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+
+// eslint-disable-next-line
+import process from 'process';
+
+interface IChatFeedback {
+  [key: number]: Feedback;
+}
+
+function ChatAssistantApp() {
+  const styles = useStyles();
+  const config = useApi(configApiRef);
+  const appThemeApi = useApi(appThemeApiRef);
+  const activeThemeId = useObservable(
+    appThemeApi.activeThemeId$(),
+    appThemeApi.getActiveThemeId(),
+  );
+  const logWithContext = (message: string) => {
+    // console.log(`[ChatAssistantApp] ${message}`);
+  };
+  const backendUrl =
+    process.env.AGENTFORGE_BACKEND_URL ||
+    config.getOptionalString('agentForge.baseUrl') ||
+    config.getString('backend.baseUrl');
+  logWithContext(`backendUrl determined: ${backendUrl}`);
+
+  if (process.env.AGENTFORGE_BACKEND_URL) {
+    logWithContext(
+      `Using AGENTFORGE_BACKEND_URL from environment: ${process.env.AGENTFORGE_BACKEND_URL}`,
+    );
+  } else if (config.getOptionalString('agentForge.baseUrl')) {
+    logWithContext(
+      `Using agentForge.baseUrl from config: ${config.getOptionalString(
+        'agentForge.baseUrl',
```

Comment:
I think we are going to run into CORS errors with this. I'm wondering if it would make sense to use the Backstage Proxy plugin to help make this easier: https://backstage.io/docs/tutorials/using-backstage-proxy-within-plugin/